src/cache.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * LRU Cache - Least Recently Used Cache
+ * 
+ * Simple cache implementation using Map.
+ * 
+ */
+export type RemoveCallbackType<T> = (key: string, value: T) => Promise<void>;
+
+export class LRUCache<T> {
+  private cache: Map<string, T>;
+
+  constructor(
+    private capacity: number,
+    private removeCallback: RemoveCallbackType<T> = async () => { }) {
+    this.capacity = capacity;
+    this.cache = new Map();
+  }
+
+  get(key: string) {
+    if (this.cache.has(key)) {
+      const value = this.cache.get(key);
+      if (value) {
```

Comment:
delete and reset on every get is not great from perf perspective.
If you want a faster implementation, you can check out how it's done in https://github.com/kibertoad/toad-cache/blob/main/src/LruMap.js

Reply:
Yeah, it seems like delete+set would be slow, but I have been surprised so many times in the past where the seemly inefficient solution happens to be fast due to JS runtime internals... I would rather make some benchmarks first before changing this to a more complex solution. The int could also be the case that this solution allows 1M get calls per second, meaning that a faster "get" will not lead to a noticeable faster "addJob".

Reply:
I've measured difference between toad-cache and tinylru. tinylru does explicit `set` for bumping on every get, while toad-cache doesn't. Perf difference is 3.8K ops/sec vs 4.2K ops/sec. I expect that explicit delete on top should make it significantly slower - delete is a pretty expensive operation in JavaScript: https://stackoverflow.com/questions/27397034/why-is-delete-slow-in-javascript

Reply:
Maybe "bun" is super fast, but when testing this:
```ts
import { LRUCache } from './src/cache';

const cache = new LRUCache<string>(3);

const start = Date.now();
for (let i = 0; i < 1_000_000; i++) {
    cache.put(`key${i}`, `value${i}`);
}

console.log(`Put: ${Date.now() - start}ms`);

for (let i = 0; i < 1_000_000; i++) {
    cache.get(`key${i}`);
}

console.log(`Get: ${Date.now() - start}ms`);
```

I get this results:
```bash
$ bun bench-cache.ts
Put: 284ms
Get: 428ms
```
So around 2.3M ops/sec

Reply:
If I increase the cache size to 3000:
```bash
$ bun bench-cache.ts
Put: 291ms
Get: 482ms
```

Reply:
nice! yeah, looks like bun is super-optimized for this type of use-cases